lib/ansiblelint/__init__.py

```diff
@@ -50,53 +50,76 @@ def matchlines(self, file, text):
         matches = []
         if not self.matchtask:
             return matches
+
         yaml = ansiblelint.utils.parse_yaml_linenumbers(text, file['path'])
-        if yaml:
-            for task in ansiblelint.utils.get_normalized_tasks(yaml, file):
-                if 'action' in task:
-                    result = self.matchtask(file, task)
-                    if result:
-                        message = None
-                        if isinstance(result, six.string_types):
-                            message = result
-                        taskstr = "Task/Handler: " + ansiblelint.utils.task_to_str(task)
-                        matches.append(Match(task[ansiblelint.utils.LINE_NUMBER_KEY], taskstr,
-                                       file['path'], self, message))
+        if not yaml:
+            return matches
+
+        for task in ansiblelint.utils.get_normalized_tasks(yaml, file):
+            if 'skipped_rules' in task and self.id in task['skipped_rules']:
+                continue
+
+            if 'action' not in task:
+                continue
+            result = self.matchtask(file, task)
+            if not result:
+                continue
+
+            message = None
+            if isinstance(result, six.string_types):
+                message = result
+            taskstr = "Task/Handler: " + ansiblelint.utils.task_to_str(task)
+            matches.append(Match(task[ansiblelint.utils.LINE_NUMBER_KEY], taskstr,
+                           file['path'], self, message))
         return matches
 
     def matchyaml(self, file, text):
         matches = []
         if not self.matchplay:
             return matches
+
         yaml = ansiblelint.utils.parse_yaml_linenumbers(text, file['path'])
-        if yaml and hasattr(self, 'matchplay'):
-            if isinstance(yaml, dict):
-                yaml = [yaml]
-            for play in yaml:
-                result = self.matchplay(file, play)
-                if result:
-                    if isinstance(result, tuple):
-                        result = [result]
-
-                    if not isinstance(result, list):
-                        raise Exception("{} is not a list".format(result))
-
-                    for section, message in result:
-                        matches.append(Match(play[ansiblelint.utils.LINE_NUMBER_KEY],
-                                             section, file['path'], self, message))
+        if not yaml:
+            return matches
+
+        if isinstance(yaml, dict):
+            yaml = [yaml]
+
+        for play in yaml:
+            if 'skipped_rules' in play and self.id in play['skipped_rules']:
+                continue
+
+            result = self.matchplay(file, play)
+            if not result:
+                continue
+
+            if isinstance(result, tuple):
+                result = [result]
+
+            if not isinstance(result, list):
+                raise Exception("{} is not a list".format(result))
```

Comment:
We need to be more precise for exceptions to be useful:
```suggestion
                raise TypeError("{} is not a list".format(result))
```